71.md

```diff
@@ -0,0 +1,35 @@
+NIP-71
+======
+
+NUDs
+----
+
+`draft` `optional`
+
+This NIP defines the creation of **NUD**s: _Nostr Unofficial Documents_, which are descriptions of standards and sub-standards that do not pertain to this NIPs repo.
+
+Anyone can create a NUD and they are immediatelly valid as soon as they are published. NUDs have owners and can be modified by these owners at any time, but they can also be forked or reinterpreted by others. However as implementations of these specs mature they will naturally coalesce into some accepted definition and it won't be possible for anyone to change that anymore in practice (even if they change the document, the Schelling point will just move to some other document or other set of documents).
+
+NUDs are defined as [NIP-54](54.md) **kind:30818** events.
+
+They should have a `d` tag starting with `"nud:"` followed by a succinct and identifiable name so they can be referred to in other contexts.
```

Comment:
I think this is a mistake. Having drafted a nud, I find myself at a loss of how to refer to it. `NUD: custom feeds` is unwieldy and potentially ambiguous. What if the `d` tag was a random hex number? We could then refer to NUDs by the first `n` digits of the hex, for example `NUD f38`. This would also accommodate situations where the name changes, which happens pretty commonly with NIPs.

Reply:
This is what FEPs do, but I don't know where the hex comes from. https://codeberg.org/fediverse/fep

Reply:
Yes, but the purpose of a wiki is to define a term. The only term involved in a NUD is the NUD itself, since the articulation of the purpose of the NUD may change. This is the same thing going on as with kind numbers and NIP numbers, which is a genius aspect of the protocol.

Reply:
@alexgleason Apparently it's the hash of the title, which does not solve the problem. Why not just pick a random 4 digit hex string? I guess `65536` are enough NUDs no? Also by restricting it to be outside the 1-99 range, we could keep calling them NIPs (I'm also with @vitorpamplona and @staab on this).

Reply:
Yeah, 256 bits is overkill. What about 6 characters, which gives us 16 million possible NUDs? Too many is better than too few.

Reply:
1. Numbers are easier than hexes. 
2. Keep the range small. 
3. Numbers should be granted to you by a simple DVM. The same DVM keeps a list of all NUDs to assemble the NUD website. 

If the DVM goes crazy, we just create a new one and regrant all past numbers.

Reply:
> I think this is a mistake. Having drafted a nud, I find myself at a loss of how to refer to it.

Why? Just say "my custom feeds NUD" or something like that, no?

If you're going to use numbers you'll risk colliding and then we'll need a central registry that guarantees exclusivity of numbers. The idea of NUDs was that anyone could pick any name they want and collisions will be impossible because each NUD will have a single creator/authority behind it.

Reply:
> Why? Just say "my custom feeds NUD" or something like that, no?

NIPs get renamed all the time, it would be nice to be able to rename a NUD but keep references intact. You'll also have collisions if there end up being two distinct NUDs that are named the same thing (e.g. "NUD: Read Receipt"). That would make it impossible to fork both NUDs under the same pubkey.

> If you're going to use numbers you'll risk colliding and then we'll need a central registry that guarantees exclusivity of numbers.

Hence my suggestion to use more randomness, this buys you:

- A shortcut to being able to talk about NUDs in an unambiguous way (you'd still have to qualify by pubkey if you're talking about forks)
- Enough randomness to avoid collisions and the need for a registry
- The ability to trace NUD identity through renames/forks

Reply:
How do you deal with name changes then?

Reply:
For context, wikipedia also deals with name changes like https://en.wikipedia.org/wiki/Internet_celebrity that people want to change to https://en.wikipedia.org/wiki/Influencer (for now, Influencer redirects to Internet_celebrity).

If you ask me, I'd change `d-tag` from `nud:orange-cars` to `orange-car` and point the article to the `nud` article using something like [these r tags](https://github.com/arthurfranca/nips/blob/trust-rank/54.md#closely-related-article-tags). See, what makes it a NUD, is the reference to the NUD article.

Heheh everyone wants to do it differently

Reply:
@staab 

> NIPs get renamed all the time, it would be nice to be able to rename a NUD but keep references intact.

What @arthurfranca said.

> You'll also have collisions if there end up being two distinct NUDs that are named the same thing (e.g. "NUD: Read Receipt").

That's why NUDs are tied to a unique owner. So we'll have Bob's Read Receipt and Carol's Read Receipt.

> That would make it impossible to fork both NUDs under the same pubkey.

Yes, but it makes no sense to fork your own NUD. You can just edit it, or make a new thing with a slightly different name, be creative.

> Hence my suggestion to use more randomness

Someone can just pretend to be generating random numbers and collide with all your NUDs on purpose, now what?

Reply:
> Yes, but it makes no sense to fork your own NUD.

There might be a collision between two NUDs belonging to other people that you want to fork.

> Someone can just pretend to be generating random numbers and collide with all your NUDs on purpose, now what?

Replaceable events were a mistake, we should just use `e` tags and build a tree.

Reply:
You can't build a tree, you can only plant a tree.